frontend/packages/kubevirt-plugin/src/components/vms/vm-activity.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import DashboardCard from '@console/shared/src/components/dashboard/dashboard-card/DashboardCard';
+import DashboardCardBody from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardBody';
+import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
+import DashboardCardLink from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardLink';
+import DashboardCardTitle from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardTitle';
+import ActivityBody, {
+  RecentEventsBody,
+} from '@console/shared/src/components/dashboard/activity-card/ActivityBody';
+import { getName, getNamespace } from '@console/shared';
+import { resourcePath, FirehoseResource, FirehoseResult } from '@console/internal/components/utils';
+import { EventModel } from '@console/internal/models';
+import { EventKind } from '@console/internal/module/k8s';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboard/with-dashboard-resources';
+import { VMKind } from '../../types';
+import { VirtualMachineModel } from '../../models';
+import { getVmEventsFilters } from '../../selectors/event';
+import { VMDashboardContext } from './vm-dashboard-context';
+
+const combinedVmFilter = (vm: VMKind) => {
+  const filters = getVmEventsFilters(vm);
+
+  return (event: EventKind): boolean =>
+    filters.some((filter) => {
+      const { name, namespace, kind } = event.involvedObject;
+      return filter({ name, namespace, kind });
+    });
+};
+
+const getEventsResource = (namespace: string): FirehoseResource => ({
+  isList: true,
+  kind: EventModel.kind,
+  prop: 'events',
+  namespace,
+});
+
+const RecentEvent = withDashboardResources(
+  ({ watchK8sResource, stopWatchK8sResource, resources, vm }: RecentEventProps) => {
+    React.useEffect(() => {
+      if (vm) {
+        const eventsResource = getEventsResource(getNamespace(vm));
+        watchK8sResource(eventsResource);
+        return () => {
+          stopWatchK8sResource(eventsResource);
+        };
+      }
+      return null;
+    }, [watchK8sResource, stopWatchK8sResource, vm]);
+    return (
+      <RecentEventsBody
```

Comment:
there shouldnt be `Recent events` title according to design

Reply:
This will require change to shared code. To keep this PR simple, I prefer to implement it in a follow-up.

Reply:
Follow-up: https://github.com/openshift/console/pull/3039